src/main/java/spark/Request.java

```diff
@@ -388,7 +388,7 @@ public Session session() {
      * <code>create</code> is <code>false</code> and the request has no valid session
      */
     public Session session(boolean create) {
-        if (session == null) {
+        if (session == null || session.raw() == null) {
```

Comment:
this "session == null || session.raw() == null" part is duplicated. And I would suggest another simple method which explains what it means (eg : isSessionRawNull() )